hoomd/write/gsd_burst.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2009-2023 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Write GSD last :math:`N` frames at user direction."""
+
+from hoomd import _hoomd
+from hoomd.filter import All
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.write.gsd import GSD
+
+
+class Burst(GSD):
+    r"""Write last :math:`N` stored frames at user trigger in the GSD format.
+
+    This class stores the last :math:`N` frames in an interal deque
+    which only writes the frames out when `dump` is called. When the writer is
+    triggered and the next frame would result in :math:`N + 1` frames being
+    stored, the oldest frame is popped off the deque and the new frame is
+    pushed to the front of the deque.
```

Comment:
```suggestion
    stored, the oldest frame is removed from the deque and the new frame is
    added.
```
Also, consider replacing `deque` (an implementation detail) with a more generic term such as `queue` or `buffer` to help users understand.